pkg/ui/workspaces/cluster-ui/src/databases/combiners.ts

```diff
@@ -84,3 +92,73 @@ const deriveDatabaseDetails = (
     numIndexRecommendations,
   };
 };
+
+interface DerivedTableDetailsParams {
+  dbName: string;
+  tables: string[];
+  tableDetails: Record<string, TableDetailsState>;
+  nodeRegions: Record<string, string>;
+  isTenant: boolean;
+}
+
+export const deriveTableDetailsMemoized = createSelector(
+  (params: DerivedTableDetailsParams) => params.dbName,
+  (params: DerivedTableDetailsParams) => params.tables,
+  (params: DerivedTableDetailsParams) => params.tableDetails,
+  (params: DerivedTableDetailsParams) => params.nodeRegions,
+  (params: DerivedTableDetailsParams) => params.isTenant,
+  (
+    dbName,
+    tables,
+    tableDetails,
+    nodeRegions,
+    isTenant,
+  ): DatabaseDetailsPageDataTable[] => {
+    tables = tables || [];
+    return tables.map(table => {
+      const tableID = generateTableID(dbName, table);
+      const details = tableDetails[tableID];
+      return deriveDatabaseTableDetails(table, details, nodeRegions, isTenant);
+    });
+  },
+);
+
+const deriveDatabaseTableDetails = (
+  table: string,
+  details: TableDetailsState,
+  nodeRegions: Record<string, string>,
+  isTenant: boolean,
+): DatabaseDetailsPageDataTable => {
+  const results = details?.data?.results;
+  const grants = results?.grantsResp.grants || [];
```

Comment:
nit: prefer `??` over `||` when checking for nullish.

Reply:
Done.